Fix origin being included in per viewport grid visibility toggle, and add ability to hide it seperately

ryevdokimov opened the pull request:
<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

* Related: https://github.com/godotengine/godot-proposals/issues/11363

https://github.com/godotengine/godot/pull/88584 added the ability to hide the grid per viewport, but accidentally included the origin, which is technically a bug. This PR separates them out, and now both can be toggled. 

https://github.com/user-attachments/assets/6bb07d6f-6705-402e-9e4e-d9539dfd831c

ryevdokimov committed:
Add ability to hide origin per viewport

aaronfranke commented:
When the origin is not drawn, the grid lines are supposed to be drawn in its place.

ryevdokimov commented:
>When the origin is not drawn, the grid lines are supposed to be drawn in its place.

Good catch, I modified the code so that the origin lines have a higher render priority so that they are always on top of the grid lines and removed the need to have to rebuild the grid.

badsectoracula commented:
The PR seems to work as expected, but IMO this is more of a new feature than a bug fix - since essentially it adds the option to toggle the grid and origin separately, adding to the UI of an already overloaded menu - so i think it is a good idea to make a feature suggestion for this and link this PR from it.

ryevdokimov commented:
There is a proposal linked above that effectively covers that https://github.com/godotengine/godot-proposals/issues/11363.  I agree as well that the menu is overloaded, but I do believe there is another proposal to clean that up.
